src/wp-includes/rest-api/endpoints/class-wp-rest-registered-templates-controller.php

```diff
@@ -1,6 +1,11 @@
 <?php
 
 class WP_REST_Registered_Templates_Controller extends WP_REST_Templates_Controller {
+	public function __construct() {
+		$this->rest_base = 'registered-templates';
+		$this->namespace = 'wp/v2';
+	}
```

Comment:
PhpStorm is complaining about this not calling `parent::__construct()`:

<img width="848" height="337" alt="Image" src="https://github.com/user-attachments/assets/3f773e81-edd3-4466-b6d2-5722e089519d" />

Maybe this class shouldn't be extending `WP_REST_Templates_Controller` but rather extend `WP_REST_Controller`? 

Or should this be passing `wp_temlate` as the post type?

```suggestion
	public function __construct() {
		$this->rest_base = 'registered-templates';
		$this->namespace = 'wp/v2';
		parent::__construct( 'wp_template' );
	}
```

Reply:
Otherwise, there are methods in` WP_REST_Templates_Controller` which are expecting `$this->post_type` to be set, but it is `null` at present.

Reply:
That's true, though I think none of these methods will be called. And yes, ideally we refactor this to not extend `WP_REST_Templates_Controller` at all. Perhaps I can just call the parent construct for now and look at this after Beta 2.

Reply:
Though `parent::__construct( 'wp_template' );` will have to be called before setting the other properties or it will override those.

Reply:
Added the parent construct call. I'll avoid extending `WP_REST_Templates_Controller` in a follow-up.